eks/deploy.go

```diff
@@ -57,112 +58,157 @@ func RollOutDeployment(
-		logger.Errorf("Error while cordoning nodes.")
-		logger.Errorf("Continue to cordon nodes that failed manually, and then terminate the underlying instances to complete the rollout.")
-		return err
+	if !state.DetachInstancesDone {
+		logger.Infof("Removing old nodes from ASG %s: %s", eksAsgName, strings.Join(state.ASG.OriginalInstances, ","))
+		err = detachInstances(asgSvc, eksAsgName, state.ASG.OriginalInstances)
+		if err != nil {
+			logger.Errorf("Error while detaching the old instances.")
+			logger.Errorf("Either resume with the recovery file or continue to detach the old instances and then terminate the underlying instances to complete the rollout.")
+			return err
+		}
+		state.DetachInstancesDone = true
+		state.persist()
 	}
-	logger.Infof("Successfully cordoned old instances in cluster ASG %s", eksAsgName)
 
-	logger.Infof("Draining Pods on old instances in cluster ASG %s", eksAsgName)
-	err = drainNodesInAsg(ec2Svc, kubectlOptions, asgInfo.currentInstanceIDs, drainTimeout, deleteLocalData)
-	if err != nil {
-		logger.Errorf("Error while draining nodes.")
-		logger.Errorf("Continue to drain nodes that failed manually, and then terminate the underlying instances to complete the rollout.")
-		return err
+	if !state.TerminateInstancesDone {
+		logger.Infof("Terminating old nodes: %s", strings.Join(state.ASG.OriginalInstances, ","))
+		err = terminateInstances(ec2Svc, state.ASG.OriginalInstances)
+		if err != nil {
+			logger.Errorf("Error while terminating the old instances.")
+			logger.Errorf("Either resume with the recovery file or continue to terminate the underlying instances to complete the rollout.")
+			return err
+		}
+		logger.Infof("Successfully removed old nodes from ASG %s", eksAsgName)
+		state.TerminateInstancesDone = true
+		state.persist()
 	}
-	logger.Infof("Successfully drained all scheduled Pods on old instances in cluster ASG %s", eksAsgName)
 
-	logger.Infof("Removing old nodes from ASG %s", eksAsgName)
-	err = detachInstances(asgSvc, eksAsgName, asgInfo.currentInstanceIDs)
-	if err != nil {
-		logger.Errorf("Error while detaching the old instances.")
-		logger.Errorf("Continue to detach the old instances and then terminate the underlying instances to complete the rollout.")
-		return err
+	if !state.RestoreCapacityDone {
+		err := setAsgMaxSize(asgSvc, eksAsgName, state.ASG.OriginalMaxCapacity)
+		if err != nil {
+			logger.Errorf("Error while restoring ASG %s max size to %v.", state.ASG.Name, state.ASG.OriginalMaxCapacity)
+			logger.Errorf("Either resume with the recovery file or adjust ASG max size manually to complete the rollout.")
+			return err
+		}
 	}
-	err = terminateInstances(ec2Svc, asgInfo.currentInstanceIDs)
+
+	err = state.delete()
 	if err != nil {
-		logger.Errorf("Error while terminating the old instances.")
-		logger.Errorf("Continue to terminate the underlying instances to complete the rollout.")
-		return err
+		logger.Errorf("Error deleting state file %s: %s", stateFile, err.Error())
```

Comment:
I think this should be logged as a warning, and then include another log indicating that the user should manually remove the state file.